packages/flutter/lib/src/material/chip.dart

```diff
@@ -10,154 +10,1371 @@ import 'package:flutter/painting.dart';
+        super(key: key);
+
+  /// A widget to display prior to the chip's label.
+  ///
+  /// Typically a [CircleAvatar] widget.
+  final Widget avatar;
+
+  /// The primary content of the chip.
+  ///
+  /// Typically a [Text] widget.
+  final Widget label;
+
+  /// The style to be applied to the chip's label.
+  ///
+  /// This only has effect on widgets that respect the [DefaultTextStyle],
+  /// such as [Text].
+  final TextStyle labelStyle;
+
+  /// The padding around the [label] widget.
+  final EdgeInsetsGeometry labelPadding;
+
+  /// The icon displayed when [onDeleted] is non-null.
+  ///
+  /// This has no effect when [onDeleted] is null since no delete icon will be
+  /// shown.
+  ///
+  /// Defaults to an [Icon] widget containing [Icons.cancel].
+  ///
+  /// May not be null.
+  final Widget deleteIcon;
+
+  /// Called when the user taps the delete button to delete the chip.
+  ///
+  /// If null, the delete button will not appear on the chip.
+  final VoidCallback onDeleted;
+
+  /// Color for delete icon, the default being black.
+  ///
+  /// This has no effect when [onDeleted] or [deleteIcon] are null since no
+  /// delete icon will be shown.
+  ///
+  /// If [deleteIcon] is set to something other than its default, then this
+  /// will have no effect, since the color specified in the [deleteIcon] widget
+  /// will take precedence.
+  final Color deleteIconColor;
+
+  /// Message to be used for the chip delete button's tooltip.
+  ///
+  /// This has no effect when [onDeleted] or [deleteIcon] are null since no
+  /// delete icon will be shown.
+  final String deleteButtonTooltipMessage;
+
+  /// The border to draw around the chip.
+  ///
+  /// Defaults to a [StadiumBorder]. May not be null.
+  final ShapeBorder border;
+
+  /// Color to be used for the chip's background, the default is based on the
+  /// ambient [IconTheme].
+  ///
+  /// This color is used as the background of the container that will hold the
+  /// widget's label.
+  final Color backgroundColor;
+
+  /// The padding between the contents of the chip and the outside [border].
+  final EdgeInsetsGeometry padding;
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMaterial(context));
+    return new RawChip(
+      key: key,
+      avatar: avatar,
+      label: label,
+      labelStyle: labelStyle,
+      labelPadding: labelPadding,
+      deleteIcon: deleteIcon,
+      onDeleted: onDeleted,
+      deleteIconColor: deleteIconColor,
+      deleteButtonTooltipMessage: deleteButtonTooltipMessage,
+      tapEnabled: false,
+      border: border,
+      backgroundColor: backgroundColor,
+      padding: padding,
+      isEnabled: true,
+    );
+  }
+}
+
+/// A material design input chip.
+///
+/// Input chips represent a complex piece of information, such as an entity
+/// (person, place, or thing) or conversational text, in a compact form.
+///
+/// Input chips can be selected by setting [onSelected], deleted by setting
```

Comment:
I don't think you mean that input chips can be selected by setting their onSelected callback.  And etc.